squad/api/ci.py

```diff
@@ -96,9 +96,14 @@ def watch_job(request, group_slug, project_slug, version, environment_slug):
     if testjob_id is None:
         return HttpResponseBadRequest("testjob_id is required")
 
+    definition = backend.get_job_definition(testjob_id)
+    if not definition:
```

Comment:
There is another backend, [Tuxsuite](https://github.com/Linaro/squad/blob/master/squad/ci/backend/tuxsuite.py), and even though it **does not** have the concept of job definition, it still uses `/api/watchjob/` quite often. Returning a bad response wouldn't fit Tuxsuite use cases.

What do you think of this:
```diff
diff --git a/squad/api/ci.py b/squad/api/ci.py
index f59cd658..bb60adc0 100644
--- a/squad/api/ci.py
+++ b/squad/api/ci.py
@@ -96,14 +96,9 @@ def watch_job(request, group_slug, project_slug, version, environment_slug):
     if testjob_id is None:
         return HttpResponseBadRequest("testjob_id is required")
 
-    definition = backend.get_job_definition(testjob_id)
-    if not definition:
-        return HttpResponseBadRequest(f"No job definition found for job {testjob_id}: {definition}")
-
     # create TestJob object
     test_job = TestJob(
         backend=backend,
-        definition=definition,
         target=project,
         target_build=build,
         environment=environment_slug,
@@ -111,6 +106,16 @@ def watch_job(request, group_slug, project_slug, version, environment_slug):
         job_id=testjob_id
     )
 
+    # check if backend is able to fetch job definition
+    try:
+        definition = backend.get_job_definition(testjob_id)
+        if not definition:
+            return HttpResponseBadRequest(f"No job definition found for job {testjob_id}")
+        test_job.definition = definition
+    except NotImplementedError:
+        # backend does not work with job definitions
+        pass
+
     # sanitize job_url
     try:
         backend.get_implementation().job_url(test_job)
```

Reply:
Yes, this makes sense. I have included the suggested change in commit [6eeec58](https://github.com/Linaro/squad/pull/1123/commits/6eeec58434b4aefa98c452645925716cc9197bd6)